src/lib.rs

```diff
@@ -111,6 +273,118 @@ fn macho_translate_section_name(section_name: &str) -> Vec<u8> {
     name
 }
 
+fn macho_get_symbols<'a>(macho: &mach::MachO<'a>) -> Vec<Symbol<'a>> {
+    // Determine section kinds and end addresses.
+    // The section kinds are inherited by symbols in those sections.
+    // The section end addresses are needed for calculating symbol sizes.
+    let mut section_kinds = Vec::new();
+    let mut section_ends = Vec::new();
+    for segment in &macho.segments {
+        for section in segment {
+            if let Ok((section, _)) = section {
+                let sectname = section
+                    .name()
+                    .map(str::as_bytes)
+                    .unwrap_or(&section.sectname[..]);
+                let segname = section
+                    .segname()
+                    .map(str::as_bytes)
+                    .unwrap_or(&section.segname[..]);
+                let (section_kind, symbol_kind) = if segname == b"__TEXT" && sectname == b"__text" {
+                    (SectionKind::Text, SymbolKind::Text)
+                } else if segname == b"__DATA" && sectname == b"__data" {
+                    (SectionKind::Data, SymbolKind::Data)
+                } else if segname == b"__DATA" && sectname == b"__bss" {
+                    (SectionKind::UninitializedData, SymbolKind::Data)
+                } else {
+                    (SectionKind::Other, SymbolKind::Unknown)
+                };
+                let section_index = section_kinds.len();
+                section_kinds.push((section_kind, symbol_kind));
+                section_ends.push(Symbol {
+                    kind: SymbolKind::Section,
+                    section: section_index + 1,
+                    section_kind: Some(section_kind),
+                    global: false,
+                    name: &[],
+                    address: section.addr + section.size,
+                    size: 0,
+                });
+            } else {
+                // Add placeholder so that indexing works.
+                section_kinds.push((SectionKind::Unknown, SymbolKind::Unknown));
+            }
+        }
+    }
+
+    let mut symbols = Vec::new();
+    for sym in macho.symbols() {
+        if let Ok((name, nlist)) = sym {
+            // Skip STAB debugging symbols.
+            if nlist.n_type & 0xe0 != 0 {
```

Comment:
What is this magic constant? Would be good to name it.

Reply:
N_STAB, I need to add it to goblin, but I added a comment here for now.